### PR TITLE
Update FlutterPluginRegistrxxx docs for iOS

### PR DESCRIPTION
**Pull request by mravn-google**
Fixes https://github.com/flutter/flutter/issues/17703

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h [hunk 1]
```diff
@@ -24,11 +24,19 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol FlutterPlugin <NSObject>
 @required
 /**
- Registers this plugin.
+ Registers this plugin with a plugin registry.
+
+ The plugin registry itself is represented during registration by the provided
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h]: I had trouble parsing this sentence. From reading this, I'm not sure if the registry is the linkage between APIs and handlers held inside the registrar or if it is the registrar.

File: shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h [hunk 2]
```diff
@@ -24,11 +24,19 @@ NS_ASSUME_NONNULL_BEGIN
...
+ registrar, decoupling the registry API (managing a collection of registered
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h]: If there's a relationship between the registry and the registration (e.g. the registry holds a number of registrations or outcomes of), perhaps point that out explicitly.

File: shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h [hunk 3]
```diff
@@ -159,6 +167,11 @@ NS_ASSUME_NONNULL_BEGIN
 
 /**
  Registration context for a single `FlutterPlugin`.
+
+ When called upon to register itself with a plugin registry, the plugin
+ is given a `FlutterPluginRegistrar` which provides the context for that
+ registration. The registrar protocol decouples the plugin from the registry
+ itself, narrowing the API depended upon by plugins.
  */
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h]: You described the relationship of the plugin to the registry and to the registrar here. Perhaps also add a quick line for the relationship between the registry and the registrar themselves. i.e. how does registrar perform the registration on the registry? Does registrar has-a registry or vice versa?

**Comment by mravn-google** [comment]
Thanks for the review. I have attempted to clarify these matters, so PTAL.

A main point of the current design is to remain abstract and silent on the exact relationship between registrar and registry, leaving it to the implementation to ensure that each plugin gets the contextual information it needs and receives the callbacks it signs up for.

We really want to avoid coupling plugins to implementation details (such as whether or not a registrar knows its underlying registry) because plugins are beyond our control once written and distributed.

**Comment by xster** [comment]
Much clearer. Thanks for the revisions!

LGTM!

**Pull request merged**